### ISSUE TITLE
addon-a11y not working with React in Storybook 4.1.11

### ISSUE THREAD

**Issue by afisher88** [storybooks/storybook#5543]
**Describe the bug**
The A11y extension always returns 0 errors 0 passes even when there are violations

**To Reproduce**
https://github.com/afisher88/react-storybook-sandbox 

This repository contains a component that should fail on colour contrast and a component that contains a button with no type or text.

**Expected behavior**
A11y errors and passes should show in the A11y panel

**Screenshots**
![storybook](https://user-images.githubusercontent.com/5710575/52556245-ea1b6200-2de3-11e9-978c-aa874332369c.png)

**Code snippets**
**package.json**
```
  "devDependencies": {
    "@babel/core": "^7.0.1",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-react": "^7.0.0",
    "@storybook/addon-a11y": "^4.1.11",
    "@storybook/addon-actions": "^4.1.11",
    "@storybook/addon-links": "^4.1.11",
    "@storybook/addons": "^4.1.11",
    "@storybook/react": "^4.1.11",
    "babel-loader": "^8.0.2",
    "css-loader": "^2.1.0",
    "mini-css-extract-plugin": "^0.5.0",
    "node-sass": "^4.9.3",
    "path": "^0.12.7",
    "sass-loader": "^7.1.0",
    "webpack": "^4.19.0",
    "webpack-cli": "^3.1.0"
  }
```

**addons.js**
```
import '@storybook/addon-actions/register';
import '@storybook/addon-links/register';
import '@storybook/addon-a11y/register';
```

**config.js**
```
import { configure, addDecorator } from '@storybook/react';
import { checkA11y } from '@storybook/addon-a11y';

// automatically import all files ending in *.stories.js
const req = require.context('../stories', true, /.stories.js$/);

addDecorator(checkA11y);

function loadStories() {
  req.keys().forEach(filename => req(filename));
}

configure(loadStories, module);
```

**System:**
 - OS: Windows10
 - Device: Desktop
 - Browser: Chrome
 - Framework: React
 - Addons: A11y
 - Version: 4.1.11

**Additional context**
I am trying to use the global decorator, however I've tried the component decorator and have the same issue.

**Comment by leoyli** [comment]
I think this one was down for a while... But it got fixed at v5.0.0-alpha.X.

**Comment by thenathanjones** [comment]
Is it likely that this will make it into 4.x or we going to have to do a major version upgrade to get the fix?

**Comment by tmeasday** [comment]
Can confirm in official SB: https://storybooks-official.netlify.com/?selectedKind=Addons%7Ca11y&selectedStory=Invalid%20contrast&full=0&addons=1&stories=1&panelRight=0&addonPanel=%40storybook%2Faddon-a11y%2Fpanel.

@leoyli any idea which PR it was that fixed it? Can we backport it?

**Comment by ximxim** [comment]
Looking forward to having this addon work with React.

**Comment by shilman** [comment]
@tmeasday @leoyli Might be this one?

https://github.com/storybooks/storybook/pull/5101

**Comment by leoyli** [comment]
Sorry have no clues. I'm not really sure when it was fixed and have the chance to look very close into this addon...

**Comment by shilman** [comment]
@igor-borisov Do you know whether your work in #5101 would also fix this addon for 4.x?

**Comment by igor-borisov** [comment]
@shilman yup it should fix, at least you can set context for a11y checking.

**Comment by shilman** [comment]
Ok I'll figure this out if I have time or if the 5.0 release looks like it will be delayed beyond first week of March

**Comment by praneybehl** [comment]
I'm in the same boat, updated the addons to v4.1.12, but it didn't fix it. Then based on @shilman 's comment updated to @storybook/addon-a11y@v4.2.0-alpha.11 which kinda fixed it but not completely. 
Although the a11y checks are now back but switching to another story doesn't refresh it.

**Comment by shilman** [comment]
@praneybehl wanna try upgrading to the 5.0-rc? We're doing releases every few days there so if you have any problems with it, you're much more likely to get timely help there. We're all hands on deck to get that out bug-free, so any testing from the community is much appreciated!!

**Comment by praneybehl** [comment]
Thanks for the prompt reply @shilman I'm going to try v5.0-rc in about an hour and post an update if itt fixes the issue.

**Comment by praneybehl** [comment]
@shilman I bumped all storybook addons and packages to v5.0.0-rc.3. It seems to resolve the issue with  A11y addon but I'm getting loads of warning in the console:
![image](https://user-images.githubusercontent.com/2144887/53145914-4c4f4080-35f6-11e9-83c4-1d607738f7d2.png)
It appears the webpack aliases that I had to resolve fonts, are not working anymore after the update
Deps:
![image](https://user-images.githubusercontent.com/2144887/53145945-76a0fe00-35f6-11e9-95c9-a4238872b26a.png)

**Comment by shilman** [comment]
@praneybehl Awesome, glad to hear it worked for you! I'm going to close this issue for now. Mind opening a new issue for the fonts problem in 5.0?

**Comment by afisher88** [comment]
Is there an ETA for v5 release? I think i'd rather hold fire for a release than upgrading to RC for production.

**Comment by shilman** [comment]
@afisher88 We're targeting March 4 for the release

**Comment by praneybehl** [comment]
Thanks @shilman will do that.
